### DISCUSSION TITLE
Error: PrismaClient is unable to run in this browser environment, or has been bundled for the browser (running in `unknown`).

### DISCUSSION THREAD

**Discussion by Narayana212** [prisma/prisma#22540] [Q&A]
### Bug description

<img width="960" alt="Screenshot 2023-12-29 104521" src="https://github.com/prisma/prisma/assets/114774469/5f0eae47-be34-4472-94a3-0546b5d9fbc9">

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
import { PrismaClient } from '@prisma/client'

declare global {
  // eslint-disable-next-line no-var
  var cachedPrisma: PrismaClient
}

let prisma: PrismaClient
if (process.env.NODE_ENV === 'production') {
  prisma = new PrismaClient()
} else {
  if (!global.cachedPrisma) {
    global.cachedPrisma = new PrismaClient()
  }
  prisma = global.cachedPrisma
}

export const db = prisma
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```
5.7.1
```

**Comment by ludralph** [comment]
Hi @Narayana212 👋 

Thank you for raising this question.

According to the error message, you are trying to use Prisma on frontend which is not possible. Either use it in next js api routes: https://nextjs.org/docs/api-routes/introduction or use it in getServerSideProps or getStaticProps: https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering. You should note that Route Handlers are just the name for API routes when using the App directory. 

You can also take a look at this [issue](https://github.com/prisma/prisma/issues/6219) to see how other users resolved the same issue.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
